### PR TITLE
data: preload persons and export

### PR DESCRIPTION
**Pull request by rerowep**
* Adds export cli.
* Adds the possibility to preload persons

## Why are you opening this PR?

- to preload persons

## How to test?

- run setup with option -P and no MEF connection

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

File: scripts/setup [hunk 1]
```diff
@@ -81,7 +82,7 @@ ES_MAPPING=false
 msg "PROGRAM: ${PROGRAM}"
 
 # options may be followed by one colon to indicate they have a required argument
-if ! options=$(getopt -o dsb -l -t deployment,create_items_holdings_small,create_items_holdings_big,lazy,pursue,data_path,time,es-mapping: -- "$@")
+if ! options=$(getopt -o dsb -l -t deployment,persons,create_items_holdings_small,create_items_holdings_big,lazy,pursue,data_path,time,es-mapping: -- "$@")
```
Comment (blankoworld) [inline scripts/setup]: `P` is missing in `getopt`, ie:

```bash
getopt -o dsbP -l -t deployment,persons,[…]
```

Comment (rerowep) [reply scripts/setup]: Not only P ....

File: scripts/setup [hunk 2]
```diff
@@ -98,7 +99,7 @@ if [ "${IS_VIRTUALENV}" != 1 ]; then
 fi
 
 # options may be followed by one colon to indicate they have a required argument
-if ! options=$(getopt -o dsbwm -l -t deployment,create_items_holdings_small,create_items_holdings_big,lazy,pursue,warnings,data_path,time,es-mapping: -- "$@")
+if ! options=$(getopt -o dPsbclptmwD: -l -t deployment,persons,create_items_holdings_small,create_items_holdings_big,lazy,pursue,time,es-mapping,warnings,data_path: -- "$@")
```
Comment (blankoworld) [inline scripts/setup]: I think we can change `-l -t` to `-l` as `-t` doesn't exists as argument in `getopt` command. What do you think?

File: scripts/setup [hunk 3]
```diff
@@ -273,10 +275,21 @@ then
...
     HOLDINGS=${DATA_PATH}/holdings_big.json
+    PERSONS=${DATA_PATH}/persons_big.json
 else
     DOCUMENTS=${DATA_PATH}/documents_small.json
     ITEMS=${DATA_PATH}/items_small.json
     HOLDINGS=${DATA_PATH}/holdings_small.json
+    PERSONS=${DATA_PATH}/persons_small.json
+fi
+
+if $LOADPERSONS
```
Comment (blankoworld) [inline scripts/setup]: As you change `$DEPLOYMENT` to `${DEPLOYMENT}` before, is that possible to do the same thing with `$LOADPERSONS`?

File: scripts/setup (hunk 2 above)
Comment (rerowep) [reply scripts/setup]: Will try, can not find any help pages for this.

File: scripts/setup (hunk 3 above)
Comment (rerowep) [reply scripts/setup]: ok

File: rero_ils/modules/cli.py [hunk 4]
```diff
@@ -1352,3 +1352,89 @@ def dump_es_mappings(verbose, outfile):
...
+@click.option('-v', '--verbose', 'verbose', is_flag=True, default=False)
+@click.option('-p', '--pid_type', 'pid_type', default='doc')
+@click.option('-o', '--outfile', 'outfile', type=click.File('w'),
+              default=sys.stdout)
+@click.option('-i', '--pidfile', 'pidfile', type=click.File('r'),
+              default=None)
+@click.option('-I', '--indent', 'indent', type=click.INT, default=2)
+@click.option('-s', '--schema', 'schema', is_flag=True, default=False)
+@with_appcontext
+def create(verbose, pid_type, outfile, pidfile, indent, schema):
```
Comment (jma) [inline rero_ils/modules/cli.py]: create or export?

File: rero_ils/modules/cli.py [hunk 5]
```diff
@@ -1352,3 +1352,89 @@ def dump_es_mappings(verbose, outfile):
...
+            if not schema:
+                del rec['$schema']
+                for subrec in rec:
+                    try:
+                        del rec[subrec]['$schema']
+                    except:
+                        pass
+            output = '\n' + json.dumps(rec, indent=indent)
+        except Exception as err:
+            print(err)
```
Comment (jma) [inline rero_ils/modules/cli.py]: click.echo?

File: rero_ils/modules/cli.py [hunk 6]
```diff
@@ -1352,3 +1352,89 @@ def dump_es_mappings(verbose, outfile):
...
+                    output += '\n' + msg
+                else:
+                    click.echo(msg)
+
+            outfile.write(output)
+            if count > 1:
+                outfile.write(',')
+            if not schema:
+                del rec['$schema']
+                for subrec in rec:
```
Comment (jma) [inline rero_ils/modules/cli.py]: user MEF source list in the config.py

File: rero_ils/modules/cli.py [hunk 7]
```diff
@@ -1352,3 +1352,89 @@ def dump_es_mappings(verbose, outfile):
...
+        .get('RECORDS_REST_ENDPOINTS')
+        .get(pid_type).get('record_class', Record))
+
+    if pidfile:
+        pids = pidfile
+    else:
+        pids = record_class.get_all_pids()
+
+    count = 0
+    if outfile == sys.stdout:
```
Comment (jma) [inline rero_ils/modules/cli.py]: why?

File: rero_ils/modules/cli.py (hunk 4 above)
Comment (rerowep) [reply rero_ils/modules/cli.py]: export

File: rero_ils/modules/cli.py (hunk 5 above)
Comment (rerowep) [reply rero_ils/modules/cli.py]: Yes

File: rero_ils/modules/cli.py (hunk 6 above)
Comment (rerowep) [reply rero_ils/modules/cli.py]: ok

File: rero_ils/modules/cli.py [hunk 8]
```diff
@@ -1352,3 +1352,80 @@ def dump_es_mappings(verbose, outfile):
             if outfile:
                 json.dump(mapping, outfile, indent=2)
                 outfile.write('\n')
+
+
+@utils.command('export')
+@click.option('-v', '--verbose', 'verbose', is_flag=True, default=False)
+@click.option('-p', '--pid_type', 'pid_type', default='doc')
+@click.option('-o', '--outfile', 'outfile', required=True,
```
Comment (blankoworld) [inline rero_ils/modules/cli.py]: What do you think about having a default value as `doc_export.json` or something like `document_dump.json` or `20200518_doc_export.json`?
It's not important, just an idea :smiley:

**Comment by blankoworld** [comment]
Tested without any internet connection: perfectly works!

**Pull request merged**